src/Operation/Distinct.php

```diff
@@ -175,6 +186,11 @@ private function createCommandDocument(): array
             $cmd['collation'] = (object) $this->options['collation'];
         }
 
+        if (isset($this->options['hint'])) {
+            /** @psalm-var string|array|object */
+            $cmd['hint'] = is_array($this->options['hint']) ? (object) $this->options['hint'] : $this->options['hint'];
```

Comment:
Isn't the `array` automatically converted to a BSON document? Or maybe that handles edge cases like `[]` or `[0 => 'foo']`.

Reply:
This is just for the edge case you highlighted.